dockers/docker-orchagent/msn27xx.32ports.buffers.json.j2

```diff
@@ -0,0 +1,294 @@
+        '25000_40'  : 'pg_lossless_25G_40m_profile',
+        '40000_40'  : 'pg_lossless_40G_40m_profile',
+        '50000_40'  : 'pg_lossless_50G_40m_profile',
+        '100000_40' : 'pg_lossless_100G_40m_profile',
+
+        '10000_300' : 'pg_lossless_10G_300m_profile',
+        '25000_300' : 'pg_lossless_25G_300m_profile',
+        '40000_300' : 'pg_lossless_40G_300m_profile',
+        '50000_300' : 'pg_lossless_50G_300m_profile',
+        '100000_300': 'pg_lossless_100G_300m_profile'
+        }
+-%}
+
+{# PG profiles. All profiles reffered in portconfig2profile dictionary should be declared here #}
+{# Only those which were actually used will be created in SAI #}
+{%- set pg_profiles = {
+        'pg_lossless_10G_5m_profile':   { 'xon': 18432, 'xoff': 16384, 'size': 34816, 'dynamic_th': 1 },
+        'pg_lossless_25G_5m_profile':   { 'xon': 18432, 'xoff': 16384, 'size': 34816, 'dynamic_th': 1 },
+        'pg_lossless_40G_5m_profile':   { 'xon': 18432, 'xoff': 16384, 'size': 34816, 'dynamic_th': 1 },
+        'pg_lossless_50G_5m_profile':   { 'xon': 18432, 'xoff': 16384, 'size': 34816, 'dynamic_th': 1 },
+        'pg_lossless_100G_5m_profile':  { 'xon': 18432, 'xoff': 18432, 'size': 36864, 'dynamic_th': 1 },
+
+        'pg_lossless_10G_40m_profile':  { 'xon': 18432, 'xoff': 18432, 'size': 36864, 'dynamic_th': 1 },
+        'pg_lossless_25G_40m_profile':  { 'xon': 18432, 'xoff': 21504, 'size': 39936, 'dynamic_th': 1 },
+        'pg_lossless_40G_40m_profile':  { 'xon': 18432, 'xoff': 23552, 'size': 41984, 'dynamic_th': 1 },
+        'pg_lossless_50G_40m_profile':  { 'xon': 18432, 'xoff': 23552, 'size': 41984, 'dynamic_th': 1 },
+        'pg_lossless_100G_40m_profile': { 'xon': 18432, 'xoff': 35840, 'size': 54272, 'dynamic_th': 1 },
+
+        'pg_lossless_10G_300m_profile': { 'xon': 18432, 'xoff': 30720, 'size': 49152, 'dynamic_th': 1 },
+        'pg_lossless_25G_300m_profile': { 'xon': 18432, 'xoff': 53248, 'size': 71680, 'dynamic_th': 1 },
+        'pg_lossless_40G_300m_profile': { 'xon': 18432, 'xoff': 75776, 'size': 94208, 'dynamic_th': 1 },
+        'pg_lossless_50G_300m_profile': { 'xon': 18432, 'xoff': 75776, 'size': 94208, 'dynamic_th': 1 },
+        'pg_lossless_100G_300m_profile':{ 'xon': 18432, 'xoff': 165888,'size': 184320,'dynamic_th': 1 },
+        }
+-%}
+
+{# Port configuration to cable length look-up table #}
+{# Each record describes mapping of DUT (DUT port) role and neighbor role to cable length #}
+{# Roles described in the minigraph #}
+{% set ports2cable = {
+        'ToRRouter_Server'       : '5',
+        'LeafRouter_ToRRouter'   : '40',
+        'SpineRouter_LeafRouter' : '300'
+        }
+%}
+
+{% set switch_role = DEVICE_METADATA['localhost']['type'] %}
+
+{%- macro cable_length(port_name) -%}
+    {%- set cable_len = [] -%}
+    {%- for local_port in DEVICE_NEIGHBOR -%}
+        {%- if local_port == port_name -%}
+            {%- if DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[local_port].name] -%}
```

Comment:
this condition doesn't work because DEVICE_NEIGHBOR[local_port].name is equal ARISTA07T0, for example, and corresponding key in DEVICE_NEIGHBOR_METADATA is "ARISTA07T0" (with quotes)
There was a fix in the minigraph.py in the old code: https://github.com/Azure/sonic-buildimage/pull/988/commits/8f3b658755944a294a77e51ae456d08899970f18

Reply:
Or remove quotes in minigraph generation
https://github.com/Azure/sonic-mgmt/blob/master/ansible/templates/topo/t0.j2#L192 (and other j2 files in this directory):
<Hostname>"{{ dev }}"</Hostname>

Reply:
ah yes. then i think we will need to remove the quotes in the minigraph generation.

Reply:
https://github.com/Azure/sonic-mgmt/pull/301 please check.